Source/Tools/LinuxEmulation/LinuxSyscalls/Seccomp/Dumper.cpp

```diff
@@ -0,0 +1,147 @@
+  };
+
+  auto Parse_Class_ST = [](uint32_t BPFIP, const sock_filter* Inst) {
+    auto DestName = [](sock_filter const* Inst) {
+      if (BPF_CLASS(Inst->code) == BPF_ST) {
+        return "A";
+      } else {
+        return "X";
+      }
+    };
+
+
+    LogMan::Msg::IFmt("0x{:04x}: Mem[{}] <- ST.{}", BPFIP, Inst->k, DestName(Inst));
+  };
+
+  auto Parse_Class_JMP = [](uint32_t BPFIP, const sock_filter* Inst) {
+    auto GetOp = [](sock_filter const* Inst) {
+      switch (BPF_OP(Inst->code)) {
+      case BPF_JA: return "a";
+      case BPF_JEQ: return "eq";
+      case BPF_JGT: return "gt";
+      case BPF_JGE: return "ge";
+      case BPF_JSET: return "set";
+      }
+      return "Unknown";
+    };
+
+    auto GetSrc = [](sock_filter const* Inst) {
+      switch (BPF_SRC(Inst->code)) {
+      case BPF_K: return fextl::fmt::format("0x{:x}", Inst->k);
+      case BPF_X: return fextl::fmt::format("<X>");
+      }
+      return fextl::fmt::format("Unknown");
+    };
+
+    LogMan::Msg::IFmt("0x{:04x}: JMP.{} {}, +{} (#0x{:x}), +{} (#0x{:x})", BPFIP, GetOp(Inst), GetSrc(Inst), Inst->jt, BPFIP + Inst->jt + 1,
+                      Inst->jf, BPFIP + Inst->jf + 1);
+  };
+
+
+  auto Parse_Class_RET = [](uint32_t BPFIP, const sock_filter* Inst) {
+    auto GetRetValue = [](sock_filter const* Inst) {
+      switch (BPF_RVAL(Inst->code)) {
+      case BPF_K: {
+        uint32_t RetData = Inst->k & SECCOMP_RET_DATA;
+        switch (Inst->k & SECCOMP_RET_ACTION_FULL) {
+        case SECCOMP_RET_KILL_PROCESS: return fextl::fmt::format("KILL_PROCESS.{}", RetData);
+        case SECCOMP_RET_KILL_THREAD: return fextl::fmt::format("KILL_THREAD.{}", RetData);
+        case SECCOMP_RET_TRAP: return fextl::fmt::format("TRAP.{}", RetData);
+        case SECCOMP_RET_ERRNO: return fextl::fmt::format("ERRNO.{}", RetData);
+        case SECCOMP_RET_USER_NOTIF: return fextl::fmt::format("USER_NOTIF.{}", RetData);
+        case SECCOMP_RET_TRACE: return fextl::fmt::format("TRACE.{}", RetData);
+        case SECCOMP_RET_LOG: return fextl::fmt::format("LOG.{}", RetData);
+        case SECCOMP_RET_ALLOW: return fextl::fmt::format("ALLOW.{}", RetData);
+        default: break;
+        }
+        return fextl::fmt::format("<Unknown>.{}", RetData);
+      }
+      case BPF_X: return fextl::fmt::format("<X>");
+      case BPF_A: return fextl::fmt::format("<A>");
+      }
+
+      return fextl::fmt::format("Unknown");
+    };
+
+    LogMan::Msg::IFmt("0x{:04x}: RET {}", BPFIP, GetRetValue(Inst));
+  };
+
+  LogMan::Msg::IFmt("BPF program: 0x{:x} instructions", prog->len);
+
+  for (size_t i = 0; i < prog->len; ++i) {
+    const sock_filter* Inst = &prog->filter[i];
+    const uint16_t Code = Inst->code;
+    const uint16_t Class = BPF_CLASS(Code);
+    switch (Class) {
+    case BPF_LD:
+    case BPF_LDX: Parse_Class_LD(i, Inst); break;
+    case BPF_ST:
+    case BPF_STX: Parse_Class_ST(i, Inst); break;
+    case BPF_ALU: break;
```

Comment:
Why are we dumping load/store/jumps but not alu?

Reply:
True, I didn't finish that off. Added the two handlers.